### PR TITLE
Authentication module api documentation[MAD-8832]

### PR DESCRIPTION
**Pull request by Neeraja82**
Authentication module api documentation[MAD-8832]

[MAD-8832]: https://liquidweb.atlassian.net/browse/MAD-8832?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: authentication/generate-passphrase.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Portal: Authentication
+
+## authentication:generate-passphrase
+Shows generated passphrase/password.
+
+#### Access
+authentication view
```
Comment (adrian-enspired) [inline authentication/generate-passphrase.md:7]: > anyone

File: authentication/generate-passphrase.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+#### Input
+- integer `length` (optional): length of the password to generate; within "{min}..{max}" length.
+- integer `words` (optional): words of the password to generate.
```
Comment (adrian-enspired) [inline authentication/generate-passphrase.md:11]: > minimum number of words to use.

File: authentication/generate-passphrase.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# Portal: Authentication
+
+## authentication:generate-passphrase
+Shows generated passphrase/password.
```
Comment (adrian-enspired) [inline authentication/generate-passphrase.md:4]: > Generates a secure passphrase.

File: authentication/generate-passphrase.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+#### Access
+authentication view
+
+#### Input
+- integer `length` (optional): length of the password to generate; within "{min}..{max}" length.
+- integer `words` (optional): words of the password to generate.
+
+#### Request
+```
+$ curl -i -X GET "$PORTAL_API_URL/v1/passphrase" \
```
Comment (adrian-enspired) [inline authentication/generate-passphrase.md:15]: no `-i` or `-X`

File: authentication/generate-passphrase.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  -H "Authorization: Bearer $PORTAL_API_KEY" \
```
Comment (adrian-enspired) [inline authentication/generate-passphrase.md:16]: no `Authorization`

File: authentication/generate-passphrase.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  "time_to_crack": {
+    "online_throttling_100_per_hour": "centuries",
+    "online_no_throttling_10_per_second": "centuries",
+    "offline_slow_hashing_1e4_per_second": "centuries",
+    "offline_fast_hashing_1e10_per_second": "12 years"
+  }
+}
+```
+
+**Failure Response** (not logged in, expired token, etc.): 401 Unauthorized
```
Comment (adrian-enspired) [inline authentication/generate-passphrase.md:45]: this endpoint will not return 401 or 403

File: authentication/login.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
+# Portal: Authentication
```
Comment (adrian-enspired) [inline authentication/login.md:1]: this endpoint is meant for use by the portal; we do not want to advertise it for customer use.

remove.

File: authentication/logout.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
+# Portal: Authentication
```
Comment (adrian-enspired) [inline authentication/logout.md:1]: this endpoint is meant for use by the portal; we do not want to advertise it for customer use.

remove.

File: authentication/assess-passphrase.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    "online_no_throttling_10_per_second": "3 years",
+    "offline_slow_hashing_1e4_per_second": "23 hours",
+    "offline_fast_hashing_1e10_per_second": "less than a second"
+  }
+}
+```
+
+**Failure Response** (not logged in, expired token, etc.): 401 Unauthorized
+
+**Failure Response** (insufficient permissions): 403 Forbidden
```
Comment (adrian-enspired) [inline authentication/assess-passphrase.md:46]: this endpoint will not return 401 or 403

File: authentication/forgot-password.md [hunk 10]
```diff
@@ -0,0 +1,36 @@
+# Portal: Authentication
```
Comment (adrian-enspired) [inline authentication/forgot-password.md:1]: this endpoint is meant for use by the portal; we do not want to advertise it for customer use.

remove.

File: authentication/assess-passphrase.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+#### Access
+authentication view
+
+#### Input
+- string `passphrase` (required): guess passphrase; within "{min}..{max}" length.
+
+#### Request
+```
+$ curl -i -X GET "$PORTAL_API_URL/v1/assess-passphrase" \
+  -H "Authorization: Bearer $PORTAL_API_KEY" \
```
Comment (adrian-enspired) [inline authentication/assess-passphrase.md:15]: no `Authorization`

File: authentication/recover-password.md [hunk 12]
```diff
@@ -0,0 +1,41 @@
+# Portal: Authentication
```
Comment (adrian-enspired) [inline authentication/recover-password.md:1]: this endpoint is meant for use by the portal; we do not want to advertise it for customer use.

remove.

**Review by adrian-enspired — CHANGES_REQUESTED** [review]

File: authentication/assess-passphrase.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
+# Portal: Authentication
+
+## authentication:login
```
[the remark this answers is not in the document]
Comment (Neeraja82) [reply authentication/assess-passphrase.md]: Changed, please review.

File: authentication/assess-passphrase.md [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+Shows guessed passphrase/password.
```
[the remark this answers is not in the document]
Comment (Neeraja82) [reply authentication/assess-passphrase.md]: Changed, please review.

File: authentication/assess-passphrase.md [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+
+#### Access
+authentication view
```
[the remark this answers is not in the document]
Comment (Neeraja82) [reply authentication/assess-passphrase.md]: Changed, please review.

File: authentication/assess-passphrase.md [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+
+#### Input
+- string `passphrase` (required): guess passphrase; within "{min}..{max}" length.
```
[the remark this answers is not in the document]
Comment (Neeraja82) [reply authentication/assess-passphrase.md]: Changed, please review.

File: authentication/assess-passphrase.md [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+
+#### Request
+```
+$ curl -i -X GET "$PORTAL_API_URL/v1/assess-passphrase" \
```
[the remark this answers is not in the document]
Comment (Neeraja82) [reply authentication/assess-passphrase.md]: Removed and changed, please review.

File: authentication/generate-passphrase.md (hunk 1 above)
Comment (Neeraja82) [reply authentication/generate-passphrase.md]: Changed, please review.

File: authentication/generate-passphrase.md (hunk 2 above)
Comment (Neeraja82) [reply authentication/generate-passphrase.md]: Updated, please review.

File: authentication/generate-passphrase.md (hunk 3 above)
Comment (Neeraja82) [reply authentication/generate-passphrase.md]: Changed, please review.

File: authentication/generate-passphrase.md (hunk 4 above)
Comment (Neeraja82) [reply authentication/generate-passphrase.md]: Removed, please review.

File: authentication/generate-passphrase.md (hunk 5 above)
Comment (Neeraja82) [reply authentication/generate-passphrase.md]: Removed, please review.

File: authentication/generate-passphrase.md (hunk 6 above)
Comment (Neeraja82) [reply authentication/generate-passphrase.md]: Removed, please review.

File: authentication/login.md (hunk 7 above)
Comment (Neeraja82) [reply authentication/login.md]: Removed, please review.

File: authentication/logout.md (hunk 8 above)
Comment (Neeraja82) [reply authentication/logout.md]: Removed, please review.

File: authentication/assess-passphrase.md (hunk 9 above)
Comment (Neeraja82) [reply authentication/assess-passphrase.md]: Removed, please review.

File: authentication/forgot-password.md (hunk 10 above)
Comment (Neeraja82) [reply authentication/forgot-password.md]: Removed, please review.

File: authentication/assess-passphrase.md (hunk 11 above)
Comment (Neeraja82) [reply authentication/assess-passphrase.md]: Removed, please review.

File: authentication/recover-password.md (hunk 12 above)
Comment (Neeraja82) [reply authentication/recover-password.md]: Removed, please review.

**Comment by Neeraja82** [comment]
Addressed all review suggestions, please review.

**Review by adrian-enspired — APPROVED** [review]

**Pull request merged**